charts/jaeger/values.yaml

```diff
@@ -11,6 +11,12 @@ tag: 1.16.0
 nameOverride: ""
 fullnameOverride: ""
 
+# Security Context for deployments
+securityContext:
```

Comment:
Can we have both podSecurityContext as well as container level security context? As done in helm create: https://github.com/helm/helm/blob/a963736f6675e972448bf7a5fd141628fd0ae4df/pkg/chartutil/create.go#L112-L121

Reply:
@naseemkullah Sure i can add both. Also are you ok with default values i provided in Values.yaml (wanted to confirm) ??

Reply:
I added podSecurityContext for pods and SecurityContext for the containers and made default values empty so as to not break current users use case.

Reply:
This is looking really good! Not sure on breakage/defaults as I have not tested, but allowing the user to is definitely a safe first step, good idea.

But I think we might want to nest the `securityContext` and `podSecurityContext` values per component (.e.g `.Values.collector.podSecurityContext`), the same way they each have their own affinity, tolerations and so on... but maybe I'm overthinking it. WDYT?

Reply:
IMO, usually in production one would not want their containers or pods to run with root privileges.Also as defined here, without giving root privileges one can give certain capabilities to their containers.
https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#set-capabilities-for-a-container
Since we kept SecurityContext its own object, one can add this key as an override and it should work.

For jaeger's components i don't see why collector or query would need different context than say an agent. ES or Cassandra can have different context anyways (by passing values to their subchart). So i don't see its need.

Let me what you think @naseemkullah ?

Reply:
Hmm I think we should test: If we wanted to give tightest securtiy context possible, can we try:

```yaml
podSecurityContext: {}
   fsGroup: 2000
securityContext: {}
   capabilities:
   drop:
   - ALL
  readOnlyRootFilesystem: true
  runAsNonRoot: true
  runAsUser: 1000
```

If some workloads run properly but others do not, it might be a sign that each component having its own security context values would be a good idea.

Reply:
@mehta-ankit, I agree, as discussed f2f, I think it's a good idea to have these security contexts attached to each of the components, therefore if someone wants to be more restrictive with one component over another, they can. Additionally, its more explicit and easier to understand exactly what the security context will be attached to. For example we do not support attaching the security context to ES/Cassanda, if the security context is defined at the root level, it's ambiguous which components it is attached to.

Reply:
@naseemkullah Made the change as discussed above. Can you review it again please. 👍  Thanks.

Reply:
lgtm!

Reply:
In fact let's go ahead and add as default value for all components' securityContext:

```
  runAsNonRoot: true
  runAsUser: 1000
```

That should be fine and seems like a sane default. If you would mind testing first that would be swell. We can maybe add more security tightening in a future PR but for now this would be good.

Reply:
Do we need both ?? If we say runAsUser 1000 its a non root user anyways (container would run with spcific uid). On the other hand if we say `runAsNonRoot: true`, why do we care about the uid, when it would be a non root user's uid. 🤔  Do you see my point here. ?
Also, as i said in my comment [here](https://github.com/jaegertracing/helm-charts/pull/37#discussion_r370105485) do we want to tighten the security by default, instead of giving users of jaegers chart a chance to provide values as they see fit for their use case.

Reply:
Also, in case you still want the defaults as mentioned in your [comment](https://github.com/jaegertracing/helm-charts/pull/37#discussion_r370822935) do we want these for both containerSecurityContext and podSecurityContext sections for all the components.??

Reply:
> Do we need both ?? If we say runAsUser 1000 its a non root user anyways (container would run with spcific uid). On the other hand if we say runAsNonRoot: true, why do we care about the uid, when it would be a non root user's uid. 🤔 Do you see my point here. ?

Yeah I do see what you mean, if we specify runAsUser, the runAsNonRoot might be redundant. I'm not familiar with best practices, I generally put both because I originally used https://kubesec.io/ to audit my workloads and it suggested to add both 🤷‍♂ ... But I see in another audit tool called Polaris, both would not be needed https://github.com/FairwindsOps/polaris/blob/91378c3d920d8ac79b197cc44c171e50ac953d32/CHANGELOG.md#051

So yeah probably just one of the two would suffice!

> Also, in case you still want the defaults as mentioned in your [comment](https://github.com/jaegertracing/helm-charts/pull/37#discussion_r370822935) do we want these for both containerSecurityContext and podSecurityContext sections for all the components.??

I'd put those in just the container `securityContext`  of all components, but have not but excessive thought into this. 

But the idea would be to run something like Polaris and reduce the number of warnings by default.

Reply:
Sorry for the back and forth on this, but let's leave them blank for now to get this PR in, and work on setting sane defaults in a separate PR.

Reply:
Sounds Good to me. I like the discussion on this though. I can make the other PR for sane defaults and we can continue the discussion there.
Will push the README changes and sign both commits and you can approve the PR then.